pkg/apic/apiserviceinstance.go

```diff
@@ -36,7 +36,7 @@ func (c *ServiceClient) buildAPIServiceInstance(
 			GroupVersionKind: mv1a.APIServiceInstanceGVK(),
 			Name:             name,
 			Title:            serviceBody.NameToPush,
-			Attributes:       util.MergeMapStringString(map[string]string{}, serviceBody.InstanceAttributes),
+			Attributes:       util.MergeMapStringString(serviceBody.InstanceAttributes),
```

Comment:
Only one map is being sent in, does it still need to be sent to mergemap? Same with the other places

Reply:
If the map being passed in is nil, then it will return a new map, so it avoids a nil check

Reply:
That makes sense. However if there is only 1 map we are using extra cycles to iterate over all items. Maybe throw a length check in the MergeMapStringString func, to return the map if only 1?

Reply:
I'll make new func to check for nil maps